File: Cpp/Maximum-Subarray.cpp
class Solution {
public:
    /**
     * @param nums: A list of integers
     * @return: A integer indicate the sum of max subarray
     */
    int maxSubArray(vector<int> nums) {
        // write your code here
        if (nums.size() == 0) return 0;
        int res = nums[0], cnt = 0;
        for (int i = 0; i < nums.size(); i ++) {
            cnt = max(cnt + nums[i], nums[i]);
            res = max(res, cnt);
        }
        return res;
    }
};

File: Python/Maximum-Subarray.py
class Solution:
    """
    @param nums: A list of integers
    @return: An integer denote the sum of maximum subarray
    """
    def maxSubArray(self, nums):
        # write your code here
        cur = 0
        maxsub = nums[0]
        for item in nums:
            cur = max(item, item + cur)
            maxsub = max(cur, maxsub)
        return maxsub